### PR TITLE
as soon as debug mode is enabled we allow listing of principals

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@PVince81 @icewind1991 @rullzer

### CODE REVIEW

**Comment by PVince81** [comment]
Could be dangerous though, if people are leaving debug mode on when hoping to get log info...

**Comment by LukasReschke** [comment]
> Could be dangerous though, if people are leaving debug mode on when hoping to get log info...

If they do that then they are to blame themselves. DEBUG mode will also open some other doors :speak_no_evil:

**Comment by DeepDiver1975** [comment]
> Could be dangerous though, if people are leaving debug mode on when hoping to get log info...

So .... and additional flag?

**Comment by karlitschek** [comment]
I agree with @LukasReschke DEBUG should be for debug only. Should never used in production. We need to enforce this instead of trying to workaround it and introducing another flag. :+1: from me.

**Comment by LukasReschke** [comment]
:+1:

File: apps/dav/lib/rootcollection.php [hunk 1]
```diff
@@ -9,14 +9,19 @@
 class RootCollection extends SimpleCollection {
 
 	public function __construct() {
+		$config = \OC::$server->getConfig();
```
Comment (rullzer) [inline apps/dav/lib/rootcollection.php]: Why not inject this?

Comment (DeepDiver1975) [inline apps/dav/lib/rootcollection.php]: Separate PR

Comment (rullzer) [inline apps/dav/lib/rootcollection.php]: OK :+1: then

**Pull request merged**